addon/helpers/has-route.js

```diff
@@ -0,0 +1,11 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+
+export default Helper.extend({
+  routing: service('-routing'),
```

Comment:
There is now a public router service, but I don't know if it has this method:
https://www.emberjs.com/api/ember/release/classes/RouterService

Reply:
unfortunately not the same service. so i have to rely on the private service, still.
alternatively, routes can be looked up via `this.owner.lookup('route:' + routeName)`, but afaik this only works from within components.

Reply:
If you can look it up on the owner then you can use `import { getOwner } from '@ember/application';` to get that value. I've only used it in tests, but it should serve the same purpose in a helper. Probably worth it to avoid the private routing service lookup.

Reply:
fair point. i'll give that a try.